src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/__tests__/components/base/FileUploadWithTagComponent.test.tsx

```diff
@@ -0,0 +1,307 @@
+  let mockMinNumberOfAttachments: number;
+  let mockReadOnly: boolean;
+  let mockAttachments: any[];
+  let mockInitialState: any;
+  let mockOptionsId: string;
+  let mockStore: any;
+  let mockGetTextResource: (key: string) => string;
+  let mockGetTextResourceAsString: (key: string) => string;
+  let mockTextResourceBindings: ITextResourceBindings;
+
+  beforeEach(() => {
+    const createStore = configureStore();
+    mockId = 'mockId';
+    mockAttachments = [
+      { name: 'attachment-name-1', id: 'attachment-id-1', size: '1200', uploaded: true, deleting: false, updating: false, tags: ['mock-tag-1'] },
+      { name: 'attachment-name-2', id: 'attachment-id-2', size: '800', uploaded: false, deleting: false, updating: false, tags: ['mock-tag-2'] },
+      { name: 'attachment-name-3', id: 'attachment-id-3', size: '400', uploaded: true, deleting: true, updating: false, tags: ['mock-tag-3'] },
+      { name: 'attachment-name-4', id: 'attachment-id-4', size: '200', uploaded: true, deleting: true, updating: false, tags: [] },
+      { name: 'attachment-name-5', id: 'attachment-id-5', size: '100', uploaded: true, deleting: false, updating: true, tags: [] },
+      { name: 'attachment-name-6', id: 'attachment-id-6', size: '100', uploaded: true, deleting: false, updating: false, tags: [] },
+    ];
+    mockTextResourceBindings = {
+      'tagTitle': 'mock-tag-title',
+      'mock-tag-label-1': "mock-tag-value-1",
+      'mock-tag-label-2': "mock-tag-value-2",
+      'mock-tag-label-3': "mock-tag-value-3",
+    }
+    mockInitialState = {
+      attachments: {
+        attachments: {
+          mockId: mockAttachments,
+        },
+        validationResults: {
+          mockId: {
+            simpleBinding: {
+              errors: ['mock error message', 'attachment-id-5' + String.fromCharCode(31) + 'mock error message'],
+            },
+          },
+        },
+      },
+      optionState: {
+        options: {
+          test: [
+            { value: 'mock-tag-1', label: 'mock-tag-label-1' },
+            { value: 'mock-tag-2', label: 'mock-tag-label-2' },
+            { value: 'mock-tag-3', label: 'mock-tag-label-3' }
+          ],
+        },
+      },
+      formLayout: {
+        uiConfig: {
+          fileUploadersWithTag: {
+            'mockId': {
+              editIndex: -1,
+              chosenOptions: {
+                'attachment-id-1': 'mock-tag-1',
+                'attachment-id-2': 'mock-tag-2',
+                'attachment-id-3': 'mock-tag-3',
+              }
+            }
+          }
+        }
+      }
+    };
+    mockOptionsId = 'test';
+    mockMaxNumberOfAttachments = 7;
+    mockMinNumberOfAttachments = 1;
+    mockMaxFileSizeInMB = 2;
+    mockIsValid = true;
+    mockReadOnly = false;
+    mockStore = createStore(mockInitialState);
+    mockGetTextResource = () => 'test';
+    mockGetTextResourceAsString = () => 'test';
+  });
+
+  it('+++ should match snapshot', () => {
```

Comment:
Snapshot tests of entire components (especially for large components) have limited value, and could be more harmful than helpful. Making a minor change to the component may cause a large diff in the snapshot, making it very difficult to see what "interesting parts" of the snapshot changed. From experience, I've seen that (large) snapshot diffs just gets updated, and nobody really reviews the changes... 

- [ ] Remove snapshot test

Reply:
Sure